Class Bmi has virtual member functions, but no virtual destructor

Anything inheriting from the provided `Bmi` class and being referred to through that abstract base class will run afoul of an inability to properly destroy itself and its members.

Cf https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-dtor

This results in compiler warnings such as this:

```
In file included from /Users/phil/Code/noaa/ngen/extern/test_bmi_cpp/src/test_bmi_cpp.cpp:6:
/Users/phil/Code/noaa/ngen/extern/test_bmi_cpp/include/test_bmi_cpp.hpp: In function 'void bmi_model_destroy(TestBmiCpp*)':
/Users/phil/Code/noaa/ngen/extern/test_bmi_cpp/include/test_bmi_cpp.hpp:275:17: warning: deleting object of polymorphic class type 'TestBmiCpp' which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
  275 |                 delete ptr;
```

--- a/bmi.hxx
+++ b/bmi.hxx
@@ -19,6 +19,8 @@ namespace bmi {
 
   class Bmi {
     public:
+      virtual ~Bmi() { }
+
       // Model control functions.
       virtual void Initialize(std::string config_file) = 0;
       virtual void Update() = 0;

